### PR TITLE
Initialize smart-guessr service

### PR DESCRIPTION
**Pull request by rimrakhimov**

### CODE REVIEW

File: whatsabi/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            path: '/api/v1/swagger'
+        }))
+        .use(
+            logger({
+                level: "info",
+            })
+        )
+        .get("/api/v1/abi", async ( { request, log, query }) => {
+            log.info(request, "New request");
+            return await processAbi(query.address, query.provider)
```
Comment (sevenzing) [inline whatsabi/src/index.ts:21]: maybe add some log to indicate that for `address` we found abi with N amount of entries

File: .github/workflows/whatsabi.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - uses: actions-ecosystem/action-regex-match@v2
+        id: regex
+        with:
+          text: ${{ github.ref }}
+          regex: '^(refs\/tags\/{{project-name}}\/(v\d+\.\d+\.\d+))|(refs\/heads\/(main))$'
```
Comment (sevenzing) [inline .github/workflows/whatsabi.yml:60]: `{{project-name}} -> whatsabi`

**Review by sevenzing — APPROVED** [review]

File: whatsabi/src/index.ts (hunk 1 above)
Comment (sevenzing) [reply whatsabi/src/index.ts:21]: or `address not found`
or `no bytecode for address`
and so on

File: whatsabi/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                if (!ethers.isAddress(query.address)) {
+                    return error(400, "Invalid address value")
+                }
+            }
+        })
+        .listen(port);
+}
+
+let app = initApp(port);
+export type App = typeof app;
```
Comment (lok52) [inline whatsabi/src/index.ts:40]: Unused?

File: whatsabi/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    `🦊 Elysia is running at ${app.server?.hostname}:${app.server?.port}`
+);
+
+async function processAbi(address: string, provider_url: string) {
+    const provider = ethers.getDefaultProvider(provider_url);
+    const signatureLookup = new whatsabi.loaders.MultiSignatureLookup([
+        new whatsabi.loaders.OpenChainSignatureLookup(),
+        new whatsabi.loaders.SamczunSignatureLookup(),
+        new whatsabi.loaders.FourByteSignatureLookup(),
+    ]);
```
Comment (lok52) [inline whatsabi/src/index.ts:52]: Probably could be extracted as a global const to prevent repeated object creation

**Review by lok52 — APPROVED** [review]

File: whatsabi/src/index.ts (hunk 1 above)
Comment (rimrakhimov) [reply whatsabi/src/index.ts]: Btw, in case if address is not a contract, then just empty array will be returned as a result and not an error

**Pull request merged**